core/templates/dev/head/pages/exploration_editor/simple_editor_tab/SimpleEditorBodyDirective.js

```diff
@@ -0,0 +1,128 @@
+// Copyright 2014 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for the body of the simple editor.
+ * editor.
+ */
+
+oppia.directive('simpleEditorBody', [function() {
```

Comment:
I do think this directive is doing too much. It's actually performing stateful operations, which I would sooner expect to be part of some service. What's the simplest contract needed by the simple editor body directive? If it can be simplified down to just a few functions (maybe just adding/querying questions, with some minor changes to interaction data), then the more complicated stuff (like interacting with answer groups, customization args, etc.) can be done in a separate service.

Adding a new question, for instance, pretty much does no UI stuff (except setting ``$scope.questions``). I would expect changing the structure of a simple editor exploration to be in a service and the directive just calling into that, updating ``$scope`` states as necessary. Happy to discuss this more offline, too.